promotion/src/main/java/org/cryptimeleon/incentive/promotion/promotions/Promotion.java

```diff
@@ -0,0 +1,63 @@
+package org.cryptimeleon.incentive.promotion.promotions;
+
+import org.cryptimeleon.incentive.crypto.model.PromotionParameters;
+import org.cryptimeleon.incentive.promotion.model.Basket;
+import org.cryptimeleon.incentive.promotion.reward.Reward;
+import org.cryptimeleon.math.serialization.StandaloneRepresentable;
+import org.cryptimeleon.math.structures.cartesian.Vector;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Promotion objects completely determine the semantic of a promotion, how the token looks like, for what a user can earn points,
+ * and which rewards a user can obtain.
+ */
+public interface Promotion extends StandaloneRepresentable {
+
+    /**
+     * Hacky way of requiring a static function that generates promotion parameters for this promotion class.
```

Comment:
This looks fun.
Why exactly would just writing a static function signature not have been enough?
Can static interface functions be implemented as dynamic functions in implementing classes?

Reply:
As far as I know Java does not support static interface functions, but I'd like to have a tightly coupled function for convenience. I am open for better alternatives though!